l10n_es_facturae_face/models/res_partner.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# © 2017 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import models, api, exceptions, _
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.constrains('facturae', 'vat', 'country_id', 'state_id',
+                    'invoice_integration_method_ids')
+    def constrain_face(self):
+        for record in self:
+            if record.invoice_integration_method_ids.filtered(
```

Comment:
Carga una vez solo la referencia y puedes utilizar el operador `in` y además sin necesitar el `.id`:
```
face = self.env.ref('l10n_es_facturae_face.integration_face')
for record in self.filtered(lambda x: face in x.invoice_integration_method_ids):
   ...
```
```